test/e2e/e2e.go

```diff
@@ -326,6 +354,19 @@ func WaitForLogContents(clients *test.Clients, logf logging.FormatLogger, podNam
 	})
 }
 
+// WaitForLogContentCount checks if the number of substr occur times equals the given number.
+// If the content does not appear the given times it returns error.
+func WaitForLogContentCount(client *test.Clients, podName, containerName, content string, appearTimes int) error {
```

Comment:
The logic of this function doesn't quite match its block comment or actual usage. If the pod log can be obtained, but a wrong number of occurrences of content is found, it returns {false, nil}, which the calling code in test/e2e/channel_chain_test.go or test/e2e/event_transformation_test.go will not flag as a test failure.

Reply:
According to the go document - "PollImmediate tries a condition func until it returns true, an error, or the timeout is reached."

If the function returns {false, nil}, it will retry the function and check the pod log again. And If it's always finding a wrong number of content occurrences, the `PollImmediate` function will finally return a `timeout` error and will be a test failure. So I think the function is working as intended.

Reply:
@Fredy-Z You're right, I missed the PollImmediate angle of it